### PR TITLE
Enable ssl for juju

### PR DESCRIPTION
**Pull request by mbruzek**
I made the changes necessary to enable Juju to use tls communication with the kube-apiserver.

The changes included adding code to the charms that generate self signed certificates and authentication files.  And changing cluster/juju/util.sh to use those values.

I found that a namespace was required to run some tests so the util.sh script now creates an empty namespace.

### CODE REVIEW

**Comment by chuckbutler** [comment]
@mbruzek Looking this over, it looks like you've encapsulated a lot of the fixes we've identified in the Kubernetes charms, thank you for the contribution to correct the Juju cluster directory and add SSL/login enablement. 

It does however look like you need to rebase, as its failing on shippable. This is typically due to doc headers or needing to rebase with a current MASTER release.

Once the rebase has been performed and cleans up the conflict :+1: lgtm.

**Comment by erictune** [comment]
this LGTM once it is rebased.

**Comment by mbruzek** [comment]
I rebased as requested. Please let me know if you need anything else.

File: cluster/juju/config-default.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# Copyright 2014 The Kubernetes Authors All rights reserved.
```
Comment (jlowdermilk) [inline cluster/juju/config-default.sh]: 2015.

File: cluster/juju/config-test.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# Copyright 2014 The Kubernetes Authors All rights reserved.
```
Comment (jlowdermilk) [inline cluster/juju/config-test.sh]: 2015

**Comment by jlowdermilk** [comment]
Travis is complaining about incorrect boilerplate on some of the files. I pointed out a few occurrences, but run `hack/verify-boilerplate.sh` to see them all.

**Comment by jlowdermilk** [comment]
Also, please squash commits down to one or a small number. Lots of tiny/incremental commits makes the history hard to read.

**Comment by mbruzek** [comment]
I created a new branch to squash these commit messages https://github.com/kubernetes/kubernetes/pull/14004

**Pull request closed without merging**